src/ast/passes/codegen_llvm.cpp

```diff
@@ -1534,10 +1534,14 @@ void CodegenLLVM::binop_ptr(Binop &binop)
   {
     // Cannot use GEP here as LLVM doesn't know its a pointer
     bool leftptr = binop.left->type.IsPtrTy();
-    auto &ptr = leftptr ? binop.left->type : binop.right->type;
+    auto &ptr_ty = leftptr ? binop.left->type : binop.right->type;
+    auto &other_ty = leftptr ? binop.right->type : binop.left->type;
     Value *ptr_expr = leftptr ? lhs : rhs;
     Value *other_expr = leftptr ? rhs : lhs;
-    auto elem_size = b_.getInt64(ptr.GetPointeeTy()->GetSize());
+
+    if (other_ty.GetSize() != 8)
```

Comment:
Semantic analysis made sure this is a pointer or an int. So here we're encoding that pointer size == int64_t size == 8. While this is correct, I think it would be more obvious with an explicit integer check on other_ty as well, so it doesn't look like we may zext pointers.

Reply:
Good point, added an int check.